source/agora/consensus/data/Transaction.d

```diff
@@ -31,6 +32,7 @@ public enum TxType : ubyte
 {
     Payment,
     Freeze,
+    Payload
```

Comment:
Why are you introducing a new transaction type?

If the transaction's payload array is not empty, then it's a payload transaction.

If you remove this special `Payload` TxType then you can remove all this special case code.

Reply:
Thank you for your good suggestion. 
I have just completed the modification according to your suggestion.